### PR TITLE
[Impeller] Fix validation errors in RendererTest.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Vulkan validation was tripping on the fact that renderer tests were rendering to the root render pass. This root render pass doesn't contain a stencil buffer. However, `MakeDefaultPipelineDescriptor` assumes a stencil and color attachment. The other backends are resilient to this mismatch since there is no compat render pass created upfront. But Vulkan was sad. This should only happen in the low level tests. In the higher levels of the framework, we have variants that make sure there is a pipeline pass and render pass compatibility.

Fixes validations of the kind:

```
--- Vulkan Debug Report  ----------------------------------------
|                Severity: Error
|                    Type: { Validation }
|                 ID Name: VUID-vkCmdDraw-renderPass-02684
|               ID Number: 1349015333
|       Queue Breadcrumbs: [NONE]
|  CMD Buffer Breadcrumbs: [NONE]
|         Related Objects: RenderPass [16305153808034431137] [Playground Render Pass], RenderPass [18100546345029861533] [Compat Render Pass: BoxFade Pipeline]
|                 Trigger: Validation Error: [ VUID-vkCmdDraw-renderPass-02684 ] Object 0: handle = 0xe2478b00000000a1, name = Playground Render Pass, type = VK_OBJECT_TYPE_RENDER_PASS; Object 1: handle = 0xfb320f000000009d, name = Compat Render Pass: BoxFade Pipeline, type = VK_OBJECT_TYPE_RENDER_PASS; | MessageID = 0x50685725 | vkCmdDraw: RenderPasses incompatible between active render pass w/ VkRenderPass 0xe2478b00000000a1[Playground Render Pass] and pipeline state object w/ VkRenderPass 0xfb320f000000009d[Compat Render Pass: BoxFade Pipeline] Attachment 4294967295 is not compatible with 1: The first is unused while the second is not.. The Vulkan spec states: The current render pass must be compatible with the renderPass member of the VkGraphicsPipelineCreateInfo structure specified when creating the VkPipeline bound to VK_PIPELINE_BIND_POINT_GRAPHICS (https://vulkan.lunarg.com/doc/view/1.3.224.1/mac/1.3-extensions/vkspec.html#VUID-vkCmdDraw-renderPass-02684)
-----------------------------------------------------------------

```

### CODE REVIEW

**Comment by chinmaygarde** [comment]
I also named the compat render pass as the debug report was confusing when the message said the render pass wasn't compatible with an unnamed compat pass.

**Review by matanlurey — APPROVED** [review]

File: impeller/renderer/backend/vulkan/pipeline_library_vk.cc [hunk 1]
```diff
@@ -128,6 +129,13 @@ static vk::UniqueRenderPass CreateRenderPass(const vk::Device& device,
     return {};
   }
 
+  // This pass is not used with the render pass. It is only necessary to tell
+  // Vulkan the expected render pass layout. The acutal pass will be create
```
Comment (jonahwilliams) [inline impeller/renderer/backend/vulkan/pipeline_library_vk.cc:133]:
```suggestion
  // Vulkan the expected render pass layout. The acutal pass will be created
```

File: impeller/renderer/backend/vulkan/pipeline_library_vk.cc [hunk 2]
```diff
@@ -128,6 +129,13 @@ static vk::UniqueRenderPass CreateRenderPass(const vk::Device& device,
     return {};
   }
 
+  // This pass is not used with the render pass. It is only necessary to tell
+  // Vulkan the expected render pass layout. The acutal pass will be created
```
Comment (bdero) [inline impeller/renderer/backend/vulkan/pipeline_library_vk.cc:133]:
```suggestion
  // Vulkan the expected render pass layout. The actual pass will be created
```

**Review by bdero — APPROVED** [review]

File: impeller/renderer/backend/vulkan/pipeline_library_vk.cc (hunk 2 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/pipeline_library_vk.cc:133]: Its been a team effort on this sentence 😆

**Pull request merged**